index.js

```diff
@@ -76,9 +78,34 @@ Plugin.prototype.apply = function(compiler) {
         });
         chunks[chunk.name] = files;
       });
+
+      var entryPoints = {};
+      stats.compilation.entrypoints.forEach(function(value, entrypoint) {
+      entrypointsChunks = value.chunks.map(function(chunk) {
+        var renderedHashToAppend = '-' + chunk.renderedHash;
+        var chunkName = chunk.name;
+        if (production) {
+          chunkName += renderedHashToAppend;
```

Comment:
I'm suspicious about that line. I think adding the hash should not be dependent to the environment, but to the `output` webpack setting. Unfortunately I'll not able to run that code soon, so I can't give you a suggestion how to change that.